### ISSUE TITLE
LiveReload not working in Firefox 4 due to missing WebSocket support

### ISSUE THREAD

**Issue by justmoon** [mockko/livereload#70]
Mozilla removed WebSocket support from Firefox 4:

http://hacks.mozilla.org/2010/12/websockets-disabled-in-firefox-4/

This causes LiveReload to fail with

```
LiveReload Failed to establish connection WebSocket is not defined
```

A workaround is to go to `about:config` and set `network.websocket.override-security-block` to `true`.

**Comment by NV** [comment]
It's written on [Mozilla add-ons site](https://addons.mozilla.org/en-US/firefox/addon/livereload/):

> LiveReload uses WebSockets. To make LiveReload work you have to [enable WebSockets](https://developer.mozilla.org/en/WebSockets).

**Comment by justmoon** [comment]
Ah sorry, didn't see that. Probably doesn't hurt to have this here under Closed Issues in case anyone searches for the error message.

**Comment by NV** [comment]
Would be nice to have UI that tells user to enable `network.websocket.override-security-block` if it's not. justmoon, what do you think?

**Comment by justmoon** [comment]
Hmm, sure, maybe something like this?

```
if (!WebSocket && navigator.userAgent.indexOf('Firefox/4') != -1) {
    alert("LiveReload requires WebSockets, which are disabled in Firefox 4 by default. Please see the LiveReload page on Mozilla Addons for details and a workaround.");
    return;
}
```

**Comment by NV** [comment]
1e2b1742c72ff6ce9fca658630b3b97a969c8334. I chose to tell exactly what to do rather then referring Mozilla Addons site.

Shipped in 0.6.1. https://addons.mozilla.org/firefox/addon/livereload/
